spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -99,7 +99,7 @@ public static void assertEqualsBatch(Types.StructType struct, Iterator<Record> e
         if (checkArrowValidityVector) {
           ColumnVector columnVector = batch.column(i);
           ValueVector arrowVector = ((IcebergArrowColumnVector) columnVector).vectorAccessor().getVector();
-          Assert.assertEquals("Nullability doesn't match", expectedValue == null, arrowVector.isNull(rowId));
+          Assert.assertFalse("Nullability doesn't match", expectedValue == null ^ arrowVector.isNull(rowId));
```

Comment:
Why change this?

Reply:
While stepping through the tests I realized that the case for `expectedValue != null` wasn't being tested. So I tweaked the check. slightly to cover both the cases.